### DISCUSSION TITLE
Plesk: 404 on step 2 with a Nginx server and a Portainer installation

### DISCUSSION THREAD

**Discussion by Orlblr** [FreshRSS/FreshRSS#6380] [Q&A] [answered]
### Describe the bug

Hello. I'm very new with doing hosted stuff on the web. I installed FreshRSS on a rss subdomain by going on Portainer, then adding a freshrss stack with these instructions

> version: '3.3'
> services:
>     freshrss:
>         container_name: freshrss
>         environment:
>             - PUID=1000
>             - PGID=100
>             - TZ=Europe/Paris
>         ports:
>             - '3001:80'
>         volumes:
>             - '/docker/freshrss/data:/var/www/FreshRSS/data'
>             - '/docker/freshrss/extensions:/var/www/FreshRSS/extensions'
>         restart: unless-stopped
>         image: freshrss/freshrss

I can access it with both serverip:3001 and the rss subdomain root. On the latter however, when I click on "Submit" after picking a language, I stumble upon a 404. Specifically, it happens when submitting a POST request. 
This very process however works fine using serverip:3001 so I assume it might be a nginx problem, in which case this thread might be out of place. When trying a random POST request on another page, it however resolves well

It should be noted that the 404 isn't written in the error.log file

I haven't found older threads mentioning this, just in case i'm forwarding this here in case you have a hunch

Best regards,

### To Reproduce

1. Go on the URL with FreshRSS installed
2. Select any language and click "Submit"
3. The url /i/index.php?step=0 sends a 404

### Expected behavior

Going forward with the installation process

### FreshRSS version

the latest

### Environment information

- Database version: mysql  Ver 15.1 Distrib 10.6.16-MariaDB
- PHP version: 8.2.17
- Installation type: Docker, using Portainer
- Web server type: Nginx
- Device: PC
- OS: Win10
- Browser: Chrome 123.0.6312.106

**Comment by Alkarex** [comment]
Try to check the logs of your nginx proxy and the logs of the freshrss container. Does the POST request arrives to freshrss at all?

See also https://github.com/FreshRSS/FreshRSS/blob/edge/Docker/README.md#run-in-production

**Comment by Orlblr** [comment]
Hello
The logs confirm that Nginx is the one not handling the POST request (404) : FreshRSS logs don't have a sign of the request

I looked at the documentation you sent, more specifically [Alternative reverse proxy using nginx > Hosted as domain root](https://github.com/FreshRSS/FreshRSS/blob/edge/Docker/README.md#hosted-as-domain-root) as it's what my setup is about

I should precise that I use Plesk, thus only use "Additional nginx directives" in my subdomain setting. Ideally I don't want to modify the source files and use Plesk whenever possible. 

In those "additional directives" ![6283_add_directives](https://github.com/FreshRSS/FreshRSS/assets/142622875/846d79a8-8b40-4b68-a06c-fc687a55116c)
I added the content of the location bracket in the documentation, except using `proxy_pass http://localhost:3001;` instead of `proxy_pass http://freshrss/;`

I got a hold of the nginx.conf file for my subdomain and checked : there's no upstream directive in this file and the SSL - HTTP/2 mentions are managed though Plesk's settings.

Something that I noticed is that the documentation mentioned ports 80 and 8080 when the proxy pass I use for nginx directives uses ^prt 3001. Although, since i still can access to rss.domain.com (302 to /i/?rid=66190a68d4859, which sends a 200), I don't think it's related ?

When browsing online, the closest of the error I'm facing that I found was in [this thread](https://www.reddit.com/r/freshrss/comments/jl5owy/comment/gciye7h/), but adding proxy_cookie_path had no effect

Regards

**Comment by Orlblr** [comment, marked as answer]
Closed : it was actually a conflicting use of the reverse proxy in Plesk. If anybody got the same problem : use "Docker proxy rules" in your site in Websites & Domains in Plesk
